### ISSUE TITLE
loss

### ISSUE THREAD

**Issue by zlm-ai** [Time-MoE/Time-MoE#23]
Thank you very much for the fine-tuning code you shared.
I tried to conduct fine-tuning tests using your dataset. However, at first, the loss increased abnormally and the grad_norm became NaN. After that, the loss turned into zero.
I would be very grateful if you could give me some guidance.
![20241122-090335](https://github.com/user-attachments/assets/b59413f3-d352-4c37-83cb-313833aec006)

**Comment by Maple728** [comment]
Could you provide the command you are using?

I ran the following command and did not encounter the `nan` issue:

```bash
python main.py -m TimeMoE-50M -d Time-300B
```

Alternatively, increasing the `global_batch_size` might resolve your issue:

```bash
python main.py -m TimeMoE-50M -d Time-300B --global_batch_size 128
```

Additionally, I recommend increasing the `micro_batch_size` to utilize your GPU memory to its fullest potential, which can improve GPU utility and accelerate training speed.

You can find all the hyperparameters in the [main.py](https://github.com/Time-MoE/Time-MoE/blob/main/main.py) file.

If you have any questions, please feel free to contact us.

**Comment by zlm-ai** [comment]
> Thank you for your interest in our work.
> 
> Could you provide the command you are using?
> 
> I ran the following command and did not encounter the `nan` issue:
> 
> ```shell
> python main.py -m TimeMoE-50M -d Time-300B
> ```
> 
> Alternatively, increasing the `global_batch_size` might resolve your issue:
> 
> ```shell
> python main.py -m TimeMoE-50M -d Time-300B --global_batch_size 128
> ```
> 
> Additionally, I recommend increasing the `micro_batch_size` to utilize your GPU memory to its fullest potential, which can improve GPU utility and accelerate training speed.
> 
> You can find all the hyperparameters in the [main.py](https://github.com/Time-MoE/Time-MoE/blob/main/main.py) file.
> 
> If you have any questions, please feel free to contact us.

Thank you very much for your suggestion. I solved this problem after adjusting the micro_batch_size from the default value of 1 to 16. However, it seems that increasing the micro_batch_size just improve the GPU utility and accelerate the training speed. I actually don't quite understand how the problem was solved. If you are also interested in this problem, could you help analyze it together?

**Comment by Maple728** [comment]
Hi,

Could you please share the command you're using, along with the specifics of your environment, such as package versions and GPU type? This information will be crucial for me to replicate the issue on my end.

I ran the following command with the default settings (`micro_batch_size` = 1, `global_batch_size` = 32) and trained for 2000 iterations, but I didn't encounter any `nan` issues:

```python
python main.py -m TimeMoE-50M -d Time-300B
```
